### PR TITLE
Haris dev/api authorization db

### PR DESCRIPTION
**Pull request by harry952014**

### CODE REVIEW

File: src/main/java/com/bincoding/nonprofit/controller/JwtAuthenticationController.java [hunk 1]
```diff
@@ -48,6 +49,11 @@
         return ResponseEntity.ok(new JwtResponse(token));
     }
 
+    @RequestMapping(value = "/register", method = RequestMethod.POST)
```
[the remark this answers is not in the document]
Comment (bsiddiqui92) [reply src/main/java/com/bincoding/nonprofit/controller/JwtAuthenticationController.java:52]: I have the same code at like 38..if you can change that too would be great.

File: src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+       return this.password;
+    }
+
+    @Override
+    public String getUsername() {
+        return this.username;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
```
[the remark this answers is not in the document]
Comment (harry952014) [reply src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java:59]: Any class implementing the UserDetails interface must implement these methods. So can't remove them.

File: src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class CustomUserDetails implements org.springframework.security.core.userdetails.UserDetails {
+
+    private String username ;
```
[the remark this answers is not in the document]
Comment (harry952014) [reply src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java:20]: What do you mean by "this space"?

[the remark this answers is not in the document]
Comment (bsiddiqui92) [reply src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java:20]: haha..just meant remove the space after `username` so `username;` not `username ;`. Super nit picky but my ocd just kicking in

File: src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (bsiddiqui92) [reply src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java:59]: Aah yes. That makes sense.

File: src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (harry952014) [reply src/main/java/com/bincoding/nonprofit/config/CustomUserDetails.java:20]: Ah Ok. :)

File: src/main/java/com/bincoding/nonprofit/controller/JwtAuthenticationController.java [hunk 4]
```diff
@@ -11,12 +12,12 @@
 import org.springframework.security.authentication.DisabledException;
 import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
 import org.springframework.security.core.userdetails.UserDetails;
-import org.springframework.security.core.userdetails.UserDetailsService;
-import org.springframework.web.bind.annotation.*;
+import org.springframework.web.bind.annotation.CrossOrigin;
```
Comment (bsiddiqui92) [inline src/main/java/com/bincoding/nonprofit/controller/JwtAuthenticationController.java:15]: Nice!

**Pull request merged**